tools/functional-tester/etcd-tester/failure.go

```diff
@@ -128,6 +129,33 @@ func (f *failureKillOne) Recover(c *cluster, round int) error {
 	return c.WaitHealth()
 }
 
+type failureKillLeader struct {
+	description
+	idx int
+}
+
+func newFailureKillLeader() *failureKillLeader {
+	return &failureKillLeader{
+		description: "kill leader member",
+	}
+}
+
+func (f *failureKillLeader) Inject(c *cluster, round int) error {
+	idx, leaderEndpoint := c.GetLeader()
+	f.idx = idx
+	if leaderEndpoint == "" {
+		log.Println("etcd-tester: no leader found, instead stopping", c.ClientURLs[idx])
```

Comment:
return an error. we should not stop the cluster inside the failure. the main loop is responsible for error handling.